sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -182,21 +180,13 @@ case class DataSourceResolution(
 
   }
 
-  object V1WriteProvider {
-    private val v1WriteOverrideSet =
-      conf.useV1SourceWriterList.toLowerCase(Locale.ROOT).split(",").toSet
-
+  object V1Provider {
     def unapply(provider: String): Option[String] = {
-      if (v1WriteOverrideSet.contains(provider.toLowerCase(Locale.ROOT))) {
```

Comment:
We have a bug here. The provider can be a short name or a qualified class name, and `useV1SourceWriterList` should support both. For example, if `orc` is in `useV1SourceWriterList`, then we should fallback to DS V1 if the provider is `orc` or `org.apache.spark.sql.execution.datasources.v2.orc.OrcDataSourceV2`.

This is also how we handle the v1 fallback list in `DataFrameReader/Writer`. See https://github.com/apache/spark/pull/25465/files#diff-f70bda59304588cc3abfa3a9840653f4L211

Reply:
As a result, I need to update some tests in `PlanResolutionSuite` and `DataSourceV2SQLSuite`, which use qualified class name of orc v2 to trigger v2 code path.